NEWS.md

```diff
@@ -4,6 +4,8 @@
 
 #### NEW FEATURES
 
+1. `setkey` and `setkeyv` gains new argument `update.index`. When `TRUE` provided it will retain existing indices when sorting data.table by reordering them so they are valid for newly keyed data.table. By default argument is set to `FALSE` because of overhead (relatively small but still) related to reordering indices. Closes [#1158](https://github.com/Rdatatable/data.table/issues/1158).
```

Comment:
I would expand on "relatively small but still" a bit / emphasize/expand on "reordering indices", on first read it sounds like `setindex` will be applied all over again, but the actual implementation is much smarter 👍

Reply:
@MichaelChirico feel free to push directly. You will probably find some others phrasing improvements. It would be interesting to change default to `update.index=TRUE` and see if any revdeps are failing, if there is no bug in the logic in this PR they shouldn't.